### PR TITLE
refactor: Change C style casts to C++ style (Part 4)

### PR DESCRIPTION
**Pull request by aditi-pandit**
As per the security guideline in
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es49-if-you-must-use-a-cast-use-a-named-cast

Covers the findings in velox/functions

### CODE REVIEW

**Comment by aditi-pandit** [comment]
Test failures related to https://github.com/facebookincubator/velox/pull/11721 mostly. Will rebase after that PR is submit.

File: velox/functions/prestosql/benchmarks/ArrayWriterBenchmark.cpp [hunk 1]
```diff
@@ -154,7 +154,7 @@ class ArrayWriterBenchmark : public functions::test::FunctionBenchmarkBase {
   }
 
   vector_size_t size = 1'000;
-  size_t totalItemsCount = (size) * (size + 1) / 2;
+  size_t totalItemsCount = static_cast<size_t>((size) * (size + 1) / 2);
```
[the remark this answers is not in the document]
Comment (aditi-pandit) [reply velox/functions/prestosql/benchmarks/ArrayWriterBenchmark.cpp:157]: I'm not 100% sure but I generally thought it was that way to keep the sizes of vectors bounded in Velox. But yes, being consistent with std::vector access is a good idea.

Maybe open a discussion in Velox about it ?

File: velox/functions/prestosql/aggregates/EntropyAggregates.cpp [hunk 2]
```diff
@@ -48,8 +48,8 @@ struct EntropyAccumulator {
     if (count == 0) {
       return;
     }
-    sumC_ += (double)count;
-    sumCLogC_ += (double)count * std::log(count);
+    sumC_ += static_cast<double>(count);
+    sumCLogC_ += static_cast<double>(count * std::log(count));
```
[the remark this answers is not in the document]
Comment (aditi-pandit) [reply velox/functions/prestosql/aggregates/EntropyAggregates.cpp]: Nice catch. Updated.

File: velox/functions/prestosql/aggregates/ApproxPercentileAggregate.cpp [hunk 3]
```diff
@@ -324,13 +324,17 @@ class ApproxPercentileAggregate : public exec::Aggregate {
         BaseVector::wrapInConstant(numGroups, 0, std::move(array));
     rowResult->childAt(kPercentilesIsArray) =
         std::make_shared<ConstantVector<bool>>(
-            pool, numGroups, false, BOOLEAN(), bool(percentiles_->isArray));
```
Comment (majetideepak) [inline velox/functions/prestosql/aggregates/ApproxPercentileAggregate.cpp:327]: This was not a cast. This was a constructor which could be redundant.
Same below.

Comment (majetideepak) [reply velox/functions/prestosql/aggregates/ApproxPercentileAggregate.cpp:327]: @aditi-pandit Thanks for clarifying. If the intent is to get an r-value, let's use `std::move()` or `static_cast<bool&&>`.

Comment (aditi-pandit) [reply velox/functions/prestosql/aggregates/ApproxPercentileAggregate.cpp:327]: Yes, good suggestion. Using static_cast<bool&&> is better.

**Review by majetideepak — APPROVED** [review]
Thanks, @aditi-pandit

File: velox/functions/prestosql/benchmarks/ArrayWriterBenchmark.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (czentgr) [reply velox/functions/prestosql/benchmarks/ArrayWriterBenchmark.cpp:157]: I created https://github.com/facebookincubator/velox/discussions/11816

**Review by czentgr — APPROVED** [review]
Thanks!

**Pull request closed without merging**